libraries/AP_HAL_Linux/OpticalFlow_Onboard.cpp

```diff
@@ -90,18 +77,68 @@ void OpticalFlow_Onboard::init(AP_HAL::OpticalFlow::Gyro_Cb get_gyro)
                       "video device");
     }
 
-    if (!_videoin->set_crop(left, top, crop_width, crop_height)) {
-        AP_HAL::panic("OpticalFlow_Onboard: couldn't set video crop");
+#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BEBOP
+    _pwm = new PWM_Sysfs_Bebop(BEBOP_CAMV_PWM);
+    _pwm->set_freq(BEBOP_CAMV_PWM_FREQ);
+    _pwm->enable(true);
+
+    _camerasensor = new CameraSensor_Mt9v117(HAL_OPTFLOW_ONBOARD_SUBDEV_PATH,
+                                             hal.i2c, 0x5D, MT9V117_QVGA,
+                                             BEBOP_GPIO_CAMV_NRST,
+                                             BEBOP_CAMV_PWM_FREQ);
+    if (!_camerasensor->set_format(HAL_OPTFLOW_ONBOARD_SENSOR_WIDTH,
+                                   HAL_OPTFLOW_ONBOARD_SENSOR_HEIGHT,
+                                   V4L2_MBUS_FMT_UYVY8_2X8)) {
+        AP_HAL::panic("OpticalFlow_Onboard: couldn't set subdev fmt\n");
+    }
+    _format = V4L2_PIX_FMT_NV12;
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_MINLURE
+    std::vector<uint32_t> pixel_formats;
+
+    _videoin->get_pixel_formats(&pixel_formats);
+
+    for (uint32_t px_fmt : pixel_formats) {
+        if (px_fmt == V4L2_PIX_FMT_NV12 || px_fmt == V4L2_PIX_FMT_GREY) {
+            _format = px_fmt;
+            break;
+        }
+
+        /* if V4L2_PIX_FMT_YUYV format is found we still iterate through
+         * the vector since the other formats need no conversions. */
+        if (px_fmt == V4L2_PIX_FMT_YUYV) {
+            _format = px_fmt;
+        }
+    }
+
+    if (!_format) {
+        AP_HAL::panic("OpticalFlow_Onboard: no supported format available");
```

Comment:
there's already a panic below if the format doesn't match one of the supported formats